src/Domain/SalesforceProxyRepository.php

```diff
@@ -18,35 +18,17 @@
 
     protected function logError(string $message): void
     {
-        if ($log = $this->getLogger()) {
-            $log->error($message);
-            return;
-        }
-
-        // Fallback if logger not configured
-        echo "ERROR: $message";
+        $this->logger->error($message);
```

Comment:
the logger property is already typed as Logger so if it isn't there getLogger will just crash with a TypeError it won't return a falsy value. So all this stuff is redundant.